scipy/signal/_filter_design.py

```diff
@@ -779,14 +785,15 @@ def group_delay(system, w=512, whole=False, fs=2*pi):
 
     if _is_int_type(w):
         if whole:
-            w = np.linspace(0, 2 * pi, w, endpoint=False)
+            w = xp.linspace(0, 2 * xp.pi, w, endpoint=False)
         else:
-            w = np.linspace(0, pi, w, endpoint=False)
+            w = xp.linspace(0, xp.pi, w, endpoint=False)
     else:
-        w = np.atleast_1d(w)
-        w = 2*pi*w/fs
+        w = xp.asarray(w)
+        w = xpx.atleast_nd(w, ndim=1, xp=xp)
+        w = 2 * xp.pi * w / fs
 
-    b, a = map(np.atleast_1d, system)
+    b, a, w = map(np.asarray, (b, a, w))
```

Comment:
What is preventing the following code from being converted?

Reply:
Could we not use `signal.convolve` instead?

Reply:
Exactly,  `np.convolve` is the reason. And `signal.convolve` is also cpu-only, so there's not much to gain either way unless https://github.com/data-apis/array-api-extra/issues/286 becomes a thing.

Reply:
Ah makes sense. Tbh it seems a bit odd to me to convert to `xp` then straight back to `np` again. Why not start with `xp = array_namespace(*system, w)` then leave everything in the middle as `np` (basically unchanged from the original) and then just do one `xp.asarray` at the end?